services/cli/src/utils/rebalancing_simulator.rs

```diff
@@ -728,6 +747,61 @@ impl RebalancingSimulator {
         Ok(())
     }
 
+    /// This function applies manual SOL withdraw and deposit changes equally across all top validators
+    /// It calculates the net SOL change (deposit - withdraw) for the epoch, divides it by active stake
+    /// to get a ratio, then applies that ratio divided equally among all top validators
+    fn apply_epoch_sol_changes(&mut self, current_epoch: u16) -> Result<(), CliError> {
+        let current_epoch_u64 = current_epoch as u64;
+        if let Some(epoch_sol_data) = self.sol_epoch_map.get(&current_epoch_u64) {
+            if epoch_sol_data.active_stake == 0.0 {
+                return Ok(());
+            }
+
+            let net_sol_change = epoch_sol_data.deposit_sol - epoch_sol_data.withdraw_sol;
+            let active_top_validators: Vec<String> = self
+                .top_validators
+                .iter()
+                .filter(|v| {
+                    self.validator_stake_states
+                        .get(&v.vote_account)
+                        .map(|state| state.target != 0)
+                        .unwrap_or(false)
+                })
+                .map(|v| v.vote_account.clone())
+                .collect();
+
+            if active_top_validators.is_empty() {
+                return Ok(());
+            }
+
+            let sol_amount_per_validator = net_sol_change / active_top_validators.len() as f64;
+            info!(
+                "Epoch {}: Applying SOL changes - Net change: {:.6} SOL, Per validator: {:.6} SOL",
+                current_epoch, net_sol_change, sol_amount_per_validator
+            );
+
+            for validator_account in &active_top_validators {
+                if let Some(stake_state) = self.validator_stake_states.get_mut(validator_account) {
+                    // Calculate ratio specific to this validator's active stake
+                    let validator_ratio = if stake_state.active > 0 {
+                        sol_amount_per_validator / (stake_state.active as f64)
+                    } else {
+                        0.0
+                    };
+
+                    stake_state.apply_stake_change(validator_ratio)?;
```

Comment:
nit: might be worth renaming "apply_stake_change" to "increase_activating_stake" or something along those lines. Even reading that function before, I still felt the need to double check what it's actually changing.